rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java

```diff
@@ -85,27 +80,25 @@
     @NonNull
     private final RestControllerHelper<Component> restControllerHelper;
 
+    @NonNull
+    private final ResourceListController<Component> resourceListController = new ResourceListController<>();
+
     @RequestMapping(value = COMPONENTS_URL, method = RequestMethod.GET)
     public ResponseEntity<Resources> getComponents(Pageable pageable,
-                                                                        @RequestParam(value = "name", required = false) String name,
-                                                                        @RequestParam(value = "type", required = false) String componentType,
                                                                         @RequestParam(value = "fields", required = false) List<String> fields,
+                                                                        @RequestParam Map<String,String> requestParams,
                                                                         HttpServletRequest request) throws TException, URISyntaxException, PaginationParameterException, ResourceClassNotFoundException {
 
         User sw360User = restControllerHelper.getSw360UserFromAuthentication();
+        List<Component> allComponents = componentService.getComponentsForUser(sw360User);
 
-        List<Component> allComponents = new ArrayList<>();
-        if (name != null && !name.isEmpty()) {
-            allComponents.addAll(componentService.searchComponentByName(name));
-        } else {
-            allComponents.addAll(componentService.getComponentsForUser(sw360User));
-        }
+        requestParams = restControllerHelper.removePageableParamatersFromRequestParams(requestParams);
```

Comment:
Why is this necessary? It might be easier if you could just tell me instead of me digging in the code :-)